synapse/cryotank.py

```diff
@@ -571,3 +588,610 @@ def init(self, name, conf=None, timeout=None):
         mesg = ('cryo:init', {'name': name, 'conf': conf})
         ok, retn = self.sess.call(mesg, timeout=timeout)
         return s_common.reqok(ok, retn)
+
+
+# ----
+# TODO: could index faster maybe if ingest/normalize is separate thread from writing
+# TODO:  what to do with subprops returned from getTypeNorm
+# TODO:  need a way to specify/load custom types
+
+# FIXME: improve datapath perf by precompile
```

Comment:
Does this still need to be fixed?

Reply:
Will fix the comment. I punted and made it a separate story.